ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/bonsai/BonsaiWorldStateProvider.java

```diff
@@ -78,12 +81,36 @@ public BonsaiWorldStateProvider(
     super(worldStateKeyValueStorage, blockchain, trieLogManager);
     this.bonsaiCachedMerkleTrieLoader = bonsaiCachedMerkleTrieLoader;
     this.worldStateHealerSupplier = worldStateHealerSupplier;
+    this.evmConfiguration = evmConfiguration;
     provideCachedWorldStorageManager(bonsaiCachedWorldStorageManager);
     loadPersistedState(
         new BonsaiWorldState(
             this, worldStateKeyValueStorage, evmConfiguration, defaultWorldStateConfig));
   }
 
+  @Override
+  public Optional<MutableWorldState> getMutable(
```

Comment:
it seems to be useless ? it's already in DiffBasesWorldStateProvider class

Reply:
Agreed - I suspect I needed it before the refactoring I did. I've removed it in the latest commit.